src/mon/MonMap.cc

```diff
@@ -823,9 +844,26 @@ int MonMap::init_with_dns_srv(CephContext* cct,
   }
 }
 
+void MonMap::update_cct(CephContext *cct)
```

Comment:
please don't add this to `MonMap`, it's supposed to allow us access mon_addrs instead of changing them stored otherwhere.

Reply:
@ShyamsundarR I think @tchaikov is suggesting you add this to `CephContext`. Have it pull the data out of `MonMap` rather than have `MonMap` push the data to `CephContext`.

Reply:
I was thinking MonClient will pull from MonMap and push to CephContext, which is what I had originally done (lost in github [history](https://github.com/ceph/ceph/pull/36533#discussion_r467848215)).

My intention was to repeat the above:
```
void MonClient::handle_monmap(...) {
  ...
  MonMap::list_addrvecs((std::list<entity_addrvec_t>& ls); // new function
  CephContext::set_mon_addrs(ls);
  ...
}
```

If we do the following instead, we would need to include MonMap in CephContext, which seems fine. I will attempt the one below for now, and report back in case I hit any snags due to the added includes to CephContext.
```
void MonClient::handle_monmap(...) {
  ...
  CephContext::update_cct(MonMap&);
  ...
}
```

Reply:
Addressed in the latest push.

All review comments are addressed. Please take another look at the changes. Thanks.